### PR TITLE
Media libs/partio

### PR DESCRIPTION
**Pull request by redchillipadi**

### CODE REVIEW

File: media-libs/partio/partio-1.10.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+RDEPEND="${PYTHON_DEPS}
+	media-libs/freeglut
+	sys-libs/zlib
+	virtual/opengl
+"
+DEPEND="${RDEPEND}"
+
+src_configure() {
+	local mycmakeargs=(
+		$(cmake_use_find_package doc Doxygen)
```
Comment (thesamesam) [inline media-libs/partio/partio-1.10.1.ebuild:45]: Yes, interestingly, that argument isn't here now ;)

File: media-libs/partio/partio-1.10.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		dev-texlive/texlive-bibtexextra
+		dev-texlive/texlive-fontsextra
+		dev-texlive/texlive-fontutils
+		dev-texlive/texlive-latex
+		dev-texlive/texlive-latexextra
+	)
+"
+RDEPEND="${PYTHON_DEPS}
+	media-libs/freeglut
+	sys-libs/zlib
```
Comment (thesamesam) [inline media-libs/partio/partio-1.10.1.ebuild:38]: := on anything we link against which has a subslot

File: media-libs/partio/partio-1.1.0_p20170829.ebuild [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
-PYTHON_COMPAT=( python3_6 )
+EAPI=7
+PYTHON_COMPAT=( python3_{7,8} )
```
[the remark this answers is not in the document]
Comment (redchillipadi) [reply media-libs/partio/partio-1.1.0_p20170829.ebuild:6]: I can't find anything official on required versions upstream. I will test all ebuilds with python 3.6 to 3.9 and update accordingly.

File: media-libs/partio/partio-1.10.1.ebuild (hunk 2 above)
Comment (redchillipadi) [reply media-libs/partio/partio-1.10.1.ebuild:38]: Looks like only zlib has a subslot, so I have added it for all the ebuilds. swig does not have a slot, so I have removed the :* in 1.1.0

File: media-libs/partio/partio-1.1.0_p20170829.ebuild (hunk 3 above)
[the remark this answers is not in the document]
Comment (redchillipadi) [reply media-libs/partio/partio-1.1.0_p20170829.ebuild:6]: All ebuilds for Partio compile with python 3.6 through 3.9

File: media-libs/partio/partio-1.10.1.ebuild (hunk 1 above)
Comment (redchillipadi) [reply media-libs/partio/partio-1.10.1.ebuild:45]: See above, docdir can be removed.

File: media-libs/partio/partio-1.10.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7,8} )
```
[the remark this answers is not in the document]
Comment (redchillipadi) [reply media-libs/partio/partio-1.10.1.ebuild:6]: Done

File: media-libs/partio/partio-1.1.0_p20170829.ebuild [hunk 5]
```diff
@@ -39,16 +39,16 @@ DEPEND="${RDEPEND}
...
+	cmake_src_prepare
 
 	sed -e '/ADD_SUBDIRECTORY (src\/tests)/d' -i CMakeLists.txt || die
 }
 
 src_configure() {
 	local mycmakeargs=(
-		$(cmake-utils_use_find_package doc Doxygen)
+		$(cmake_use_find_package doc Doxygen)
 		-DCMAKE_INSTALL_DOCDIR="share/doc/${PF}"
```
[the remark this answers is not in the document]
Comment (redchillipadi) [reply media-libs/partio/partio-1.1.0_p20170829.ebuild:50]: The DOCDIR seems to work when the line is removed, but without the find_package doc Doxygen the docs are always installed regardless of whether the doc USE flag is set.

File: media-libs/partio/partio-1.1.0_p20170829.ebuild [hunk 6]
```diff
@@ -39,16 +39,16 @@ DEPEND="${RDEPEND}
 PATCHES=( "${FILESDIR}/${PN}-1.1.0-Rename-partconv.patch" )
 
 src_prepare() {
-	cmake-utils_src_prepare
+	cmake_src_prepare
 
 	sed -e '/ADD_SUBDIRECTORY (src\/tests)/d' -i CMakeLists.txt || die
```
[the remark this answers is not in the document]
Comment (redchillipadi) [reply media-libs/partio/partio-1.1.0_p20170829.ebuild:44]: Thanks, I have put cmake_src_prepare at the bottom of the function.

**Comment by redchillipadi** [comment]
I don't think there will be any need to keep the old versions around here for long. Osl it the only package that depends on partio and it works with the lastest version so once everyone has had a chance to upgrade we should be able to drop everything before 1.10.

File: media-libs/partio/partio-1.1.0_p20170829.ebuild [hunk 7]
```diff
@@ -1,10 +1,10 @@
 # Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
-PYTHON_COMPAT=( python3_6 )
+EAPI=7
+PYTHON_COMPAT=( python3_{6,7,8,9} )
 
-inherit cmake-utils vcs-snapshot python-single-r1
+inherit cmake vcs-snapshot python-single-r1
```
Comment (thesamesam) [inline media-libs/partio/partio-1.1.0_p20170829.ebuild:7]: Sort?

File: media-libs/partio/partio-1.10.1.ebuild (hunk 1 above)
Comment (thesamesam) [reply media-libs/partio/partio-1.10.1.ebuild:46]: Yep :)

File: media-libs/partio/partio-1.10.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8,9} )
```
Comment (thesamesam) [inline media-libs/partio/partio-1.10.1.ebuild:6]: Just checking: the tests (if they exist) passed on all these, but if they do not exist, at least the program seemed to work (ideally with something using said bindings? or whatever)

Comment (redchillipadi) [reply media-libs/partio/partio-1.10.1.ebuild:6]: I checked that I could compile each version, and created test.bgeo using makeline (with one of the versions), and opened it using partview with each version. I can see a collection of points in a straight line. Some of the tests like makecircle complain when they use default values, but appear to give valid results when checked using partview.

Tests like testmerge and testcluster only pass if the source is still unpacked as they need src/data/*.bgeo to exist and need portage permissions. I think the tests should eventually be packaged to run only with FEATURES="test" and not be installed on the user's system.

File: media-libs/partio/partio-1.1.0_p20170829.ebuild (hunk 7 above)
Comment (redchillipadi) [reply media-libs/partio/partio-1.1.0_p20170829.ebuild:7]: Done

**Review by mgorny — APPROVED** [review]
Thank you. It looks like nice progress!

**Pull request closed without merging**